phpBB/includes/functions_download.php

```diff
@@ -41,15 +43,12 @@ function send_avatar_to_browser($file, $browser)
 	{
 		$image_dir = '';
 	}
-	$file_path = $phpbb_root_path . $image_dir . '/' . $prefix . $file;
+	$file_path = $image_dir . '/' . $prefix . $file;
 
-	if ((@file_exists($file_path) && @is_readable($file_path)) && !headers_sent())
+	if ($storage->exists($file_path) && !headers_sent())
 	{
 		header('Cache-Control: public');
 
-		$image_data = @getimagesize($file_path);
-		header('Content-Type: ' . image_type_to_mime_type($image_data[2]));
```

Comment:
indeed, without the streams it is complicated (with streams we can read the first bytes and use @marc1706 library)
We still have a solution but we need to load the whole image in memory...